default/stringtables/en.txt

```diff
@@ -2579,6 +2579,9 @@ The Save Game button is disabled, either because this is a multiplayer game in w
 GAME_MENU_REALLY_CONCEDE
 Confirm conceding? This empire will be removed from the game.
 
+GAME_MENU_CONFIRM_NOT_READY
+Confirm quit? Still revising orders.
```

Comment:
This message also displayed if orders have never been sent for the current turn, not just if they were sent and then the revise orders button was clicked.

Reply:
True. Would you prefer different wording?

Reply:
Ideally it would check whether orders have ever been sent for the turn currently being played, or if there are orders issued but not sent since the revise orders button was clicked, or of no orders have ever been sent for the current turn, or if orders were sent and revise clicked but no no additional orders were then issued, and adjust the message accordingly. Alternatively, in a multiplayer game it could say that the server is waiting for orders, or in single player it could say orders have not yet been sent. Or in all cases it could say orders have not been sent... even though if revising then orders were sent (and just the revisions haven't been sent)...

Reply:
This kind of brings me back to my original suggestion:
"Confirm quit?  Orders have not been finalized."
This appears to cover both the case in which orders have never been sent for the current turn, and the case in which orders were sent but are currently in the state of revision.

The objection was that you do not need to click the End Turn button to finalize or submit your orders.  But after thinking it through, while your orders may finalize and submit if you allow the timer to expire, it is still accurate to say that the orders have not been _finalized_ if they are currently in the unsubmitted state or the revising state.  (I think)

Reply:
The message will not be displayed in a single player game.

I would prefer a single wording for the dialog than trying to check whether the user has submitted then cancelled to start revising. It would get extra confusing when a user logs back in to the multiplayer server and it automatically starts revising. So a consistent message regardless of that state would be easier and probably better for the user to understand.

I am also fine with alleryn's suggestion "Confirm quit? Orders have not been finalized." Or if we wanted a more concise version: "Confirm quit? Orders not finalized."

Thoughts?

Reply:
> The message will not be displayed in a single player game.

Are you sure / is that true now? I can't test immediately but I think I got the message when reseigning or closing the window in a single player game...

Reply:
> > The message will not be displayed in a single player game.
> 
> Are you sure / is that true now? I can't test immediately but I think I got the message when reseigning or closing the window in a single player game...

I'm unable to reproduce this.  I only get the dialog box when i try to exit a multiplayer game.  Exiting a singleplayer game is as before.

I think the && !m_single_player_game is intended to prevent this.

While testing i did produce an error however; i'll comment below.

Reply:
Geoff, I have never had the message come up for a single player game in my tests. As alleryn stated, I explicitly check that it is not a single player game before showing the dialog box.

Reply:
Possibly I tested with a nominally multiplayer game with only one human player, which does produce the popup.

But I think the multiplayer/singleplayer distinction isn't important really...

Regardless, "Confirm quit? Orders not finalized." seems OK if you don't want to adjust for the exactly turn-send-revise-resend state.